ARIES-1023 and ARIES-1079 fixing

ggerla opened the pull request:
In this solution there are 2 PersistenceContextProvider services registered. The first is GlobalPersistenceManager, same before, but now with property org.apache.aries.jpa.context.type=context. The second one is PersistenceBundleManager with property org.apache.aries.jpa.context.type=unit. 
In this way, when the NSHandler processes the Blueprint xml file notify unit name and bundle to "context" service and properties to "unit" service. The "unit" service create a new EntityManagerFactory using these properties.

ggerla committed:
Fixed ARIES-1023 modifing namespace expected for map element

ggerla committed:
Added notification to PersistenceBundleManager to use jpa:map properties for EntityManagerFactory creation

ggerla committed:
Merge remote-tracking branch 'aries/trunk' into trunk

ggerla committed:
Added some check to avoid NullPointerException during startup

cschneider commented:
I do not really understand why you are using an additional PersistenceContextProvider. You only explain here what it does but not why. Can you perhaps open a jira issue for it and explain what is wrong/bad at the current solution and how you solve it?

In this pull request there are two smaller solutions that seems to be separate:
https://github.com/ggerla/aries/commit/35b4a04f5d3ae192013b5e48c4605af221316746
https://github.com/ggerla/aries/commit/21511b10a3c9664592b96d09491a418bb27e5cc6
I think I can already commit these if you want. Does that make sense?

ggerla commented:
Yes you are right: there 2 different problem. 
The first one is that explained in ARIES-1023 and ARIES-1079. This issue is solved with commit 
https://github.com/ggerla/aries/commit/35b4a04f5d3ae192013b5e48c4605af221316746

I open ARIES-1295 for the second one.

ggerla committed:
Merge remote-tracking branch 'aries/trunk' into trunk

ggerla committed:
Merge remote-tracking branch 'aries/trunk' into trunk

ggerla committed:
Merge remote-tracking branch 'aries/trunk' into trunk

ggerla committed:
Merge remote-tracking branch 'aries/trunk' into trunk

Conflicts:
	jpa/itests/jpa-container-itest/src/test/java/org/apache/aries/jpa/itest/AbstractJPAItest.java
	jpa/jpa-blueprint-aries/src/main/java/org/apache/aries/jpa/blueprint/aries/impl/NSHandler.java
	jpa/jpa-blueprint-aries/src/main/resources/OSGI-INF/blueprint/jpa.xml
	jpa/jpa-blueprint-aries/src/test/java/org/apache/aries/jpa/blueprint/aries/test/NSHandlerTest.java
	jpa/jpa-blueprint-testbundle/pom.xml
	jpa/jpa-blueprint-testbundle/src/main/java/org/apache/aries/jpa/blueprint/itest/JPATestBean.java
	jpa/jpa-blueprint-testbundle/src/main/resources/OSGI-INF/blueprint/injection.xml
	jpa/jpa-blueprint-testbundle/src/main/resources/OSGI-INF/persistence/persistence.xml
	jpa/jpa-container-context/src/main/java/org/apache/aries/jpa/container/context/impl/GlobalPersistenceManager.java
	jpa/jpa-container-itest/src/test/java/org/apache/aries/jpa/blueprint/aries/itest/JPAInjectionTest.java
	jpa/jpa-container/src/main/java/org/apache/aries/jpa/container/impl/Activator.java
	jpa/jpa-container/src/main/java/org/apache/aries/jpa/container/impl/EntityManagerFactoryManager.java
	jpa/jpa-container/src/main/java/org/apache/aries/jpa/container/impl/PersistenceBundleManager.java
	jpa/jpa-container/src/main/java/org/apache/aries/jpa/container/unit/impl/ManagedPersistenceUnitInfoImpl.java

ggerla committed:
- Added support for PersistenceUnit annotation
- Added tests for EntityManager and EntityManagerFactory injection

ggerla committed:
Added classes to test Supplir and em blueprint integration

ggerla committed:
Create EmfProxyFactory to manage the EMF creation

ggerla committed:
Merge remote-tracking branch 'aries/trunk' into trunk

Conflicts:
	jpa/itests/jpa-container-blueprint-testbundle/src/main/resources/OSGI-INF/blueprint/config.xml
	jpa/itests/jpa-container-itest/src/test/java/org/apache/aries/jpa/blueprint/aries/itest/BlueprintTest.java
	jpa/itests/jpa-container-itest/src/test/java/org/apache/aries/jpa/blueprint/aries/itest/BlueprintWithEmTest.java
	jpa/itests/jpa-container-itest/src/test/java/org/apache/aries/jpa/blueprint/aries/itest/BlueprintWithMethodTest.java
	jpa/itests/jpa-container-itest/src/test/java/org/apache/aries/jpa/itest/AbstractJPAItest.java
	jpa/jpa-blueprint/src/main/java/org/apache/aries/jpa/blueprint/impl/JpaBeanProcessor.java

ggerla committed:
Merge remote-tracking branch 'aries/trunk' into trunk

ggerla committed:
- Implemented ARIES-1343 scanning bean class and all super class to find annotated fields and methods
- Add integration test to verify new functionality
- Refactoring of JpaBeanProcessor to avoid duplicated code

cschneider commented:
Hi Giuseppe. The issues seem to be solved. Can you close this PR? I will try to do it with a commit but not sure if this works.
